docs/testpmd.md

```diff
@@ -0,0 +1,238 @@
+# TestPMD 
+
+[TestPMD](https://doc.dpdk.org/guides/testpmd_app_ug/index.html) is anapplication  used to test the DPDK in a packet forwarding mode, here this application run inside a pod with SRIOV VF.
+
+## Pre-requisites
+
+This workload assumes a healthy cluster is runing with SR-IOV and Performance Add-On operators. 
+
+* [SR-IOV](https://docs.openshift.com/container-platform/4.6/networking/hardware_networks/installing-sriov-operator.html) operator is required to create/assign VFs to the pods
+* [PAO](https://docs.openshift.com/container-platform/4.6/scalability_and_performance/cnf-performance-addon-operator-for-low-latency-nodes.html) is required to prepare the worker node with Hugepages for DPDK application and isolate CPUs for workloads. 
+
+### Sample SR-IOV configuration
+
+These are used to create VFs and SRIOV networks, interface might vary. Creating `SriovNetworkNodePolicy` will reboot your worker nodes one by one.
+
+```yaml
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetworkNodePolicy
+metadata:
+  name: testpmd-policy
+  namespace: openshift-sriov-network-operator
+spec:
+  deviceType: vfio-pci
+  nicSelector:
+    pfNames:
+     - ens2f1
+  nodeSelector:
+    node-role.kubernetes.io/worker: ""
+  numVfs: 20
+  resourceName: intelnics
+```
+
+```yaml
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetwork
+metadata:
+  name: testpmd-sriov-network
+  namespace: openshift-sriov-network-operator
+spec:
+  ipam: |
+    {
+      "type": "host-local",
+      "subnet": "10.57.1.0/24",
+      "rangeStart": "10.57.1.100",
+      "rangeEnd": "10.57.1.200",
+      "routes": [{
+        "dst": "0.0.0.0/0"
+      }],
+      "gateway": "10.57.1.1"
+    }
+  spoofChk: "on"
+  trust: "on"
+  resourceName: intelnics
+  networkNamespace: my-ripsaw
+
+```
+
+### Sample PAO configuration
+
+The CPU and socket information might vary, use the below as an example. Creating `PerformanceProfile` will trigger MCO to reboot worker nodes one by one.
+Please be sure to WAIT until the previous step is finished, i.e SR-IOV VF configuration. 
+
+Add label to the MCP pool before creating the performance profile.
+```sh
+# oc label mcp worker machineconfiguration.openshift.io/role=worker
+```
+
+```yaml
+---
+apiVersion: performance.openshift.io/v1
+kind: PerformanceProfile
+metadata:
+   name: testpmd-performance-profile-0
+spec:
+   cpu:
+      isolated: "6,8,10,12,14,16,18,20,22,24,26,28"
+      reserved: "0-5,7,9,11,13,15,17,19,21,23,25,27,29"
+   hugepages:
+      defaultHugepagesSize: "1G"
+      pages:
+         - size: "1G"
+           count: 16
+           node: 0
+   numa:
+      topologyPolicy: "best-effort"
+   nodeSelector:
+      node-role.kubernetes.io/worker: ""
+   realTimeKernel:
+     enabled: false
+   additionalKernelArgs:
+   - nosmt
+   - tsc=reliable
+```
+
+## Running standalone TestPMD
```

Comment:
Should it be "Running containerized TestPMD" ?

Reply:
Correct, will update